File: src/lib/tauri/commands/fs.ts
/**
 * 文件系统相关命令封装
 *
 * 实际权限范围在 src-tauri/capabilities/*.json 中声明。
 * 这里只负责调用，不负责鉴权。
 */

import { invoke } from "@tauri-apps/api/core";

export interface ReadTextFileOptions {
  path: string;
}

export const fsCommands = {
  readTextFile: (path: string): Promise<string> =>
    invoke<string>("fs_read_text_file", { path }),

  writeTextFile: (path: string, contents: string): Promise<void> =>
    invoke<void>("fs_write_text_file", { path, contents }),

  exists: (path: string): Promise<boolean> =>
    invoke<boolean>("fs_exists", { path }),
};

File: src-tauri/src/commands/fs.rs
//! 文件系统相关 command（薄壳：参数 + 调用 service + 包装错误）

use crate::error::AppResult;
use crate::services::fs_service;

#[tauri::command]
pub fn fs_read_text_file(path: String) -> AppResult<String> {
    fs_service::read_text_file(&path)
}

#[tauri::command]
pub fn fs_write_text_file(path: String, contents: String) -> AppResult<()> {
    fs_service::write_text_file(&path, &contents)
}

#[tauri::command]
pub fn fs_exists(path: String) -> AppResult<bool> {
    fs_service::exists(&path)
}
